cpp/src/buffer/buffer.cpp

```diff
@@ -92,11 +139,24 @@ std::unique_ptr<Buffer> Buffer::copy(MemoryType target, rmm::cuda_stream_view st
                     cudaMemcpyDeviceToHost,
                     stream
                 ));
-                return std::unique_ptr<Buffer>(new Buffer{std::move(ret), br});
+                auto new_buffer = std::unique_ptr<Buffer>(new Buffer{std::move(ret), br});
+
+                // The event is created here instead of the constructor because the
+                // memcpy is async, but the buffer is created on the host.
+                new_buffer->event_ = std::make_unique<Event>(stream);
```

Comment:
```suggestion
                new_buffer->event_ = std::make_shared<Event>(stream);
```

Reply:
Fixed in https://github.com/rapidsai/rapidsmpf/pull/225/commits/e2d8c9c0fe8f5b28f5334f52cf075c3df292a141 .